### ISSUE TITLE
Ability to get children of req object

### ISSUE THREAD

**Issue by josiahwiebe** [lukeed/polka#28]
I'm trying to set a variable as the value of either `req.body.token`, `req.query.token`, or `req.headers['x-access-token']` in a middleware function, but I'm running into issues.

In Express, this works:
```javascript
const app = express()

function apikey(req, res, next) {
  let userToken = req.body.token || req.query.token || req.headers['x-access-token']
  if (!token) return res.status(401).end('no token provided')
  next()
}

app.use(apikey)
```

If a token is present in the body, it uses that, and so on. 

When I try the equivalent in Polka, I'm getting a `TypeError: Cannot read property 'token' of null` When I use the [Async example](https://github.com/lukeed/polka#async-handlers) expressed in the README I actually get the same error.

**Comment by lukeed** [comment]
Ah, if you remove the `req.query.token` part it should work. Polka isn't setting `query` or `search` until after the middleware loop.

**Comment by lukeed** [comment]
No, you're right. It used to & should continue to. Just thinking the possible side effects of the current code.

**Comment by lukeed** [comment]
Wait, I'm mistaken. It _is_ setting those keys [before iterating](https://github.com/lukeed/polka/blob/master/packages/polka/index.js#L88-L89) thru middleware (line 96).

Can you show me how you're actually calling & mounting `apikey`?

**Comment by josiahwiebe** [comment]
I've tried a few approaches to debug, but here's the one I'm trying to get to work. I'm transpiling code with Babel so it's ES7. (truncated for brevity)

`index.js`
```javascript
import polka from 'polka'

polka()
  .use('users', users)
  .listen(port)
```

`middleware/apikey`
```javascript
export async function apikey(req, res, next) {
  let userToken = req.query.token || req.headers['x-access-token']
  if (userToken) {
    let user = await User.getUser(userToken)
    if (!user) send(res, 401, { success: false, error: 'Token verification failed' })
    next()
  else {
    send(res, 401, { success: false, error: 'Token verification failed' })
  }
}
```

`routes/users`
```javascript
import polka from 'polka'
import send from '@polka/send-type'
import apikey from '../middleware/token'

export default polka()
  .use(apikey)
  .get('/', async (req, res) => {
    send(res, 200, { success: true, message: 'here is a response' })
  })
```

**Comment by lukeed** [comment]
Thanks!!

I forgot that Express casts `req.query` to an object. 

Your code works (sorta) with `/users?query=123`, but it's returned as a string. It failed in all other instances because `req.query` is `null` by default, whereas Express ensures an empty object.

**Comment by josiahwiebe** [comment]
Yeah, that isn't ideal for my use case. I can write my own helper for this as a last resort, just wanted to make sure there wasn't something I was missing in Polka.

I might also be able to enforce the use of the `req.headers` token rather than a querystring. I'm just refactoring an Express app to use Polka and want to keep the transition as smooth as possible.

**Comment by lukeed** [comment]
I'm patching it now :)

**Comment by lukeed** [comment]
Available in `0.3.4` 😄
